test/examples/uct_hello_world.c

```diff
@@ -362,13 +362,24 @@ static ucs_status_t dev_tl_lookup(const cmd_args_t *cmd_args,
             for (tl_index = 0; tl_index < num_tl_resources; ++tl_index) {
                 if (!strcmp(cmd_args->dev_name, tl_resources[tl_index].dev_name) &&
                     !strcmp(cmd_args->tl_name, tl_resources[tl_index].tl_name)) {
+                    if (!(iface_p->md_attr.cap.reg_mem_types & UCS_BIT(test_mem_type))) {
+                        fprintf(stderr, "Unsupported memory type %s by "
+                                UCT_TL_RESOURCE_DESC_FMT" on %s MD\n",
+                                ucs_memory_type_names[test_mem_type],
+                                UCT_TL_RESOURCE_DESC_ARG(&tl_resources[tl_index]),
+                                component_attr.md_resources[md_index].md_name);
+                        uct_release_tl_resource_list(tl_resources);
```

Comment:
can introduce new goto label for releasing tl resources (they are also release on line 383)

Reply:
use `continue` operator instead of `goto`, we need to continue thru existing resources like we do if IFACE initialization fails